### ISSUE TITLE
Error alert for wrong chronology of date of last contact and date of report does not stop saving of data and onward navigation.

### ISSUE THREAD

**Issue by BasilKaburi** [SORMAS-Foundation/SORMAS-Project#4382]
### Bug Description
The system gives an error alert when last date of contact is a date after the date of report. Yet, the system permits saving of wrong dates and onward navigation. 
### Steps to Reproduce
1. Start creating a new contact
2. Select any date for the last day of contact
3. For the date of report, select a date earlier to the selected date of last contact in step 2.

### Expected Behavior
An error alert will show up, but you will still be able to save wrong dates and continue navigation to the next page.
### Screenshots

<img width="1036" alt="Issue_1_Permits saving of contact data despite wrong last contact date" src="https://user-images.githubusercontent.com/71414361/107889274-b7942a80-6f11-11eb-9eda-5bc8992cd842.png">
### System Details
* Device: MacBook Pro
* SORMAS version: 1.56.0
* Android version/Browser: Safari
* Server URL: https://release-international.sormas.netzlink.com/sormas-ui/#!contacts/?includeContactsFromOtherJurisdictions=false&dateFilterOption=DATE* 
* User Role: Admin and National User

**Comment by kwa20** [comment]
@bernardsilenou is this supposed to even display an error? I would assume that in e.g. family settings, the date of last contact will likely often be extended to be after the report date. If a family member turns out to be a case, other family members will be contacts and if they are in quarantine together, every new day will be the new date of last contact. We therefore might want to think about not showing the warning when multi-day contact is selected.

**Comment by bernardsilenou** [comment]
@kwa20  Yes, its suppose to display an error because the initial issue was based on the use case that the contact ended before it was reported and that during followup, contacts does not continue taking place. Based on your explanation, it make sense to include the option where by for multiple contact, the last contact date can be after the report date.
Refinement:
1. For single contact, date of last contact <= report date and also <= current date
2. For multiple contact:
    2.1 date of first contact <= report date and ( date of first contact or report date )<= current date
          AND
    2.2 date of first contact <= date of last contact  and (date of last contact) <= current date

3.  Followup until should then be recomputed as usual based on date of last contact.

@MateStrysewske @kwa20  Did I miss something?

**Comment by MateStrysewske** [comment]
@bernardsilenou @kwa20 I think your approach is not really matching what is currently in the system. A contact right now is referring to a contact instance, not about a constant contact between two persons. I think it can make sense e.g. for people living in the same household to only have one contact instance in the system that is then constantly updated, but that would not be a "multi-day contact". Also, the report date of the contact would need to be updated everytime as well, so there wouldn't be any need to allow the "date of last contact" to be after the report date.

As a result, there's nothing we need to change about the logic here in my eyes because how exactly the users are using these contacts is their own choice. As far as I'm concerned, all we have to do is make sure that the validation is actually working correctly.

What do you think?

**Comment by bernardsilenou** [comment]
@MateStrysewske @kay   
- From what I can remember,  The goal of adding multiple day contact was then to group all the single contacts in to one contact and only increase followup time. Users did not want to create many single instances of contacts since the persons are the same.
- I also though of updating the report date each time there is a new contact after the initial report date. The limitation that I see here is the fact that we would no longer have the initial report date which is not good since we would then end up with many visits before contact report date.
- In practice users do not also know that there would be a multiple contact in advance, so there is a possibility that many multiple contacts would have their last contact after the report date. So we fall in situation that the calendar time window of the multiple contact overlap with that of the followup , thus report date of contact falls in-between first contact and last contact date.
- Since it is a possibility that the last contact can be after the initial report date of report for multiple contact, I think we should do something do document all these details and keep it flexible.
-  I do not know is this is urgent or if any user has requested such a need

**Comment by MateStrysewske** [comment]
@bernardsilenou If that's the case, I feel like "multi-day contact" is a pretty weak name for what we're trying to achieve with the field. To me, it suggests that the contact instance took place over multiple days, e.g. because two persons were on a holiday together, not that it's a constant contact that is updated all the time. However, if this is its purpose, it indeed makes sense to allow setting the last contact date to a date after the report date.

**Comment by kwa20** [comment]
@MateStrysewske I think your example describes the exact context of this feature. However, contacts can also take place over multiple days after the report date, which for example happens when families are quarantined together. One family member might be a case and all others contact persons to that case. Since they're confirmed contacts that were put into quarantine already, every following day will be the new date of last contact to the case and after the report date.

**Comment by bernardsilenou** [comment]
@MateStrysewske @kwa20  Both scenarios (household or holidays people) are to be captured by this design. If someone has a better name, then its good that we change the caption.
